From: Jeff Epler <jepler@unpythonic.net>
Date: Fri, 19 Apr 2013 08:16:19 -0500
Subject: kfreebsd has /dev/urandom

---
 src/mongo/platform/random.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/mongo/platform/random.cpp
+++ b/src/mongo/platform/random.cpp
@@ -108,7 +108,7 @@
     return new WinSecureRandom();
 }
 
-#elif defined(__linux__) || defined(__sunos__) || defined(__APPLE__) || defined(__freebsd__)
+#elif defined(__linux__) || defined(__sunos__) || defined(__APPLE__) || defined(__freebsd__) || defined(__FreeBSD_kernel__)
 
 class InputStreamSecureRandom : public SecureRandom {
 public:
