--- a/SConstruct
+++ b/SConstruct
@@ -636,6 +636,10 @@
                CONFIGUREDIR=sconsDataDir.Dir('sconf_temp'),
                CONFIGURELOG=sconsDataDir.File('config.log'),
                INSTALL_DIR=installDir,
+               CFLAGS=os.environ.get("CFLAGS", "-O3").split(),
+               CXXFLAGS=os.environ.get("CXXFLAGS", "-O3").split(),
+               CPPFLAGS=os.environ.get("CPPFLAGS", "").split(),
+               LINKFLAGS=os.environ.get("LDFLAGS", "").split(),
                )
 
 if windows:
